eclass/xdg-utils.eclass

```diff
@@ -67,7 +73,8 @@ xdg_environment_reset() {
 # Updates the .desktop files database.
 # Generates a list of mimetypes linked to applications that can handle them
 xdg_desktop_database_update() {
-	local updater="${EROOT%/}${DESKTOP_DATABASE_UPDATE_BIN}"
+	[[ ${EAPI:-0} == [0123456] ]] && BROOT=${EROOT%/}
+	local updater="${BROOT%/}${DESKTOP_DATABASE_UPDATE_BIN}"
```

Comment:
I would prefer to see these absolute binary paths eliminated, and replaced with simple calls based on PATH instead.

Reply:
So assume/enforce portage to have proper path and just call `foo-updater` without an
explicit absolute binary path, then?

Reply:
Correct. You can check if they exist in PATH using the `type` built-in. For example:
```
if type update-desktop-database &>/dev/null; then
    update-desktop-database ...
fi
```